osu.Game.Rulesets.Mania/UI/Column.cs

```diff
@@ -136,6 +136,33 @@ public void OnReleased(KeyBindingReleaseEvent<ManiaAction> e)
         {
         }
 
+        // https://github.com/ppy/osu-framework/blob/49c954321c3686628b2c223670363438f88a0341/osu.Framework/Graphics/Drawable.cs#L1513-L1524
+        private T findClosestParent<T>() where T : class, IDrawable
+        {
+            Drawable cursor = this;
+
+            while ((cursor = cursor.Parent) != null)
+            {
+                if (cursor is T match)
+                    return match;
+            }
+
+            return default;
+        }
```

Comment:
Code cannot be copy-pasted like this.

Even if touching columns is deemed to be acceptable, this should be using DI or similar, not traversing parents on _every touch_. Framework side comments even emphasise this is an expensive lookup. It should not be done on every touch.